app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/local/CharacterRealm.kt

```diff
@@ -0,0 +1,19 @@
+package com.puzzlebench.clean_marvel_kotlin.data.local
+
+import io.realm.RealmObject
+import io.realm.annotations.PrimaryKey
+
+const val EMPTY_STRING = ""
+const val DEFAULT_ID = 0
+
+open class CharacterRealm(
+        @PrimaryKey
+        var id: Int? = DEFAULT_ID,
```

Comment:
this is okay? I think that you shouldn't have a default id. The Id mustn't be nullable.

Reply:
couldn`t do id not nullable. This error shows
https://stackoverflow.com/questions/49167886/kotlin-realm-class-must-declare-a-public-constructor-with-no-arguments-if-it-co
Any help?

Reply:
i think propertys should be nullable: 
https://github.com/realm/realm-java/blob/master/examples/kotlinExample/src/main/kotlin/io/realm/examples/kotlin/model/Person.kt

Reply:
done